course_discovery/apps/api/fields.py

```
@@ -65,6 +67,19 @@ def to_internal_value(self, data):
         return clean_html(data) if data else data
 
 
+class SlugRelatedTranslatableField(serializers.SlugRelatedField):
+    """ Use in place of SlugRelatedField when the slug field is a TranslatedField """
+
+    def to_internal_value(self, data):
+        full_translated_field_name = 'translations__{slug_field}'.format(slug_field=self.slug_field)
+        try:
+            return self.get_queryset().get(**{full_translated_field_name: data})
+        except ObjectDoesNotExist:
+            self.fail('does_not_exist', slug_name=self.slug_field, value=smart_text(data))
+        except (TypeError, ValueError):
+            self.fail('invalid')
```

Haven't seen self.fail before, but if this swallows the exception it might be worth clarifying what the error actually is instead of just outputting invalid.

---

Full disclosure this was copy-pasted from the SlugRelatedField code